conan/tools/gnu/pkgconfigdeps.py

```diff
@@ -60,12 +60,22 @@ def _get_requires_names(self, name, cpp_info):
         Note: CppInfo could be coming from one Component object instead of the dependency
         """
         ret = []
-        for req in cpp_info.requires:
-            pkg_name, comp_name = req.split("::") if "::" in req else (name, req)
-            # FIXME: it could allow defining requires to not direct dependencies
-            req_conanfile = self._conanfile.dependencies.host[pkg_name]
-            comp_alias_name = get_component_name(req_conanfile, comp_name)
-            ret.append(self._get_pc_name(pkg_name, comp_alias_name))
+        if cpp_info.requires:
+            for req in cpp_info.requires:
+                pkg_name, comp_name = req.split("::") if "::" in req else (name, req)
+                # FIXME: it could allow defining requires to not direct dependencies
+                req_conanfile = self._conanfile.dependencies.host[pkg_name]
+                comp_alias_name = get_component_name(req_conanfile, comp_name)
+                ret.append(self._get_pc_name(pkg_name, comp_alias_name))
+        else:
```

Comment:
I think this `else` shouldn't apply when the `cpp_info` is from a component (called from `comp_requires_names = self._get_requires_names(str(dep.ref.name), comp_cpp_info)`) 
Probably move this `else` code to (replace?) the second call to the function: `requires = self._get_requires_names(str(dep.ref.name), dep.cpp_info)`